code/modules/flufftext/Dreaming.dm

```diff
@@ -17,15 +17,18 @@
 mob/living/carbon/proc/dream()
 	dreaming = 1
 
-	spawn(0)
-		for(var/i = rand(1,4),i > 0, i--)
-			to_chat(src, "<span class='notice'><i>... [pick(POSSIBLE_DREAM_TOPICS)] ...</i></span>")
-			sleep(rand(40,70))
-			if(!IsUnconscious())
-				dreaming = 0
-				return 0
-		dreaming = 0
-		return 1
+	INVOKE_ASYNC(src, .proc/dream_async)
+
+///handles the display of dream text to the player.
+mob/living/carbon/proc/dream_async()
+	for(var/i = rand(1,4),i > 0, i--)
+		to_chat(src, "<span class='notice'><i>... [pick(POSSIBLE_DREAM_TOPICS)] ...</i></span>")
+		sleep(rand(40,70))
+		if(!IsUnconscious())
+			dreaming = 0
+			return
+	dreaming = 0
+	return
```

Comment:
4 things: 
forgot the / in front of mob
this for loop isnt defined right you should use our stand
Dreaming is a bool
This sleep causes hard deletes ping me if you want more explained, but it needs to  be another proc